base/abstractarray.jl

```diff
@@ -720,8 +725,12 @@ end
 ## copy between abstract arrays - generally more efficient
 ## since a single index variable can be used.
 
-copyto!(dest::AbstractArray, src::AbstractArray) =
+function _copyto_impl!(dest::AbstractArray, src::AbstractArray, allowshorter::Bool)
+    if !allowshorter && length(src) < length(dest)
```

Comment:
This check and the one below is actually not needed for this PR since we already allocate an array of the correct length, and the `AbstractArray` interface relies on the fact that the length is correctly declared. But I added these for consistency, so that the `allowshorter` argument (which needs to be accepted) isn't a no-op. I can drop it and add a comment explaining that if you want.